### PR TITLE
PoC: filtering samples before screening based on contig length

### PR DESCRIPTION
**Pull request by jfy133**
We have the issue that different tools have different minimum contig lengths required to be run.

In particular, `antismash` will unfortunately kill the entire pipeline (exit code 1) if no contigs in the FASTA reach it's minimum threshold (note: if there are sufficent lengths but no BGCs found, it doesn't fail so this is OK).

This PR adds a calculation and adding to meta the length of the longest contig  in the FASTA, which allows us to filter out samples going into antismash (in this proof-of-concept) that do not reach this threshold. This will speed up the pipleine as it won't run on samples that will be default fail, and in the case of antismash, not cause an entire pipeline failure.

```
ec-10 21:04:31.141 [Actor Thread 6] INFO  nextflow.processor.TaskProcessor - [dc/67ca0a] Cached process > NFCORE_FUNCSCAN:FUNCSCAN:PRODIGAL_GFF (sample_1)
Dec-10 21:04:31.149 [Actor Thread 14] WARN  nextflow.Nextflow - [nf-core/funcscan] Sample does not have any contig reaching min. length threshold. Antismash will not be run for: sample_2.
Dec-10 21:04:31.155 [Actor Thread 14] WARN  nextflow.Nextflow - [nf-core/funcscan] Sample does not have any contig reaching min. length threshold. Antismash will not be run for: sample_1.
```

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/nf-core/bioawk/bioawk.diff [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Changes in module 'nf-core/bioawk'
+--- modules/nf-core/bioawk/main.nf
++++ modules/nf-core/bioawk/main.nf
+@@ -12,7 +12,8 @@
+ 
+     output:
+     tuple val(meta), path("*.gz"), emit: output
+-    path "versions.yml"             , emit: versions
++    tuple val(meta), env(LONGEST), emit: longest
```
Comment (Darcy220606) [inline modules/nf-core/bioawk/bioawk.diff:9]: Just to understand, why do you have the (LONGEST) as `env`? Isnt it just counting the length of contigs?

**Review by Darcy220606 — APPROVED** [review]
Looks good to me! Perhaps also mention this part in the docs e.g. in `usage.md` for the AS part ?

**Review by louperelo — APPROVED** [review]
Nice solution!

File: modules/nf-core/bioawk/bioawk.diff (hunk 1 above)
Comment (jfy133) [reply modules/nf-core/bioawk/bioawk.diff:9]: I needed a way to export JUST the value of the longest contig, so I can attach it to the `meta` that can be evaluated later in the pipeline.

So by assigning it to the ENV variable, I can isolate it at the output and then add it to meta with the `map` later on. If that makes sense?

Comment (Darcy220606) [reply modules/nf-core/bioawk/bioawk.diff:9]: Oh, that makes sense now. Thx!

File: subworkflows/local/bgc.nf [hunk 2]
```diff
@@ -61,6 +61,11 @@ workflow BGC {
 
         ch_antismash_input = fna.mix(gff)
                                 .groupTuple()
+                                .filter {
+                                    meta, files ->
+                                        if ( meta.longest_contig < params.bgc_antismash_mincontiglength ) log.warn "[nf-core/funcscan] Sample does not have any contig reaching min. length threshold. Antismash will not be run for: ${meta.id}."
```
Comment (jasmezz) [inline subworkflows/local/bgc.nf:66]:
```suggestion
                                        if ( meta.longest_contig < params.bgc_antismash_mincontiglength ) log.warn "[nf-core/funcscan] Sample does not have any contig reaching min. length threshold of ${params.bgc_antismash_mincontiglength} bp. Antismash will not be run for: ${meta.id}."
```

**Pull request merged**